### ISSUE TITLE
core receive but not send openFlow PACKET_IN packet

### ISSUE THREAD

**Issue by Maoyuxing** [fp7-netide/Engine#139]
Hi, we found that the core received OpenFlow PACKET_IN pack from the shim, however, it did not send it to backend. There were some problems while handling this packet. 
We use flood-light backend and onos shim, we use mininet to create the topology, its OpenFlow version is 1.0.
we capture some log information from the karaf.log:

2015-12-05 06:25:48,592 | INFO  | 0MQ Receive Loop | ZeroMQBasedConnector             | 159 - core.connectivity - 1.1.0.SNAPSHOT | recv OpenFlow 10
2015-12-05 06:25:48,592 | INFO  | ool-13-thread-27 | FIBManager                       | 163 - core.globalfib - 1.1.0.SNAPSHOT | FIBManager received message from shim: MessageHeader [Version=VERSION_1_4,Type=OPENFLOW,Length=99,ModuleId=0,TransactionId=0,DatapathId=3]
2015-12-05 06:25:48,592 | INFO  | ool-13-thread-27 | CompositionManager               | 160 - core.caos - 1.1.0.SNAPSHOT | CompositionManager received message from shim: OpenFlowMessage [Header=MessageHeader [Version=VERSION_1_4,Type=OPENFLOW,Length=99,ModuleId=0,TransactionId=0,DatapathId=3],Type=PACKET_IN,OFMessage=OFPacketInVer10(xid=0, bufferId=4294967295, totalLen=81, inPort=2, reason=ACTION, data=[-1, -1, -1, -1, -1, -1, -34, -83, -66, -17, -70, 17, -119, 66, 2, 7, 4, 0, 0, 0, 0, 0, 4, 4, 5, 2, 0, 0, 0, 1, 6, 2, 0, 120, -2, 18, -92, 35, 5, 1, 79, 78, 79, 83, 32, 68, 105, 115, 99, 111, 118, 101, 114, 121, -2, 23, -92, 35, 5, 2, 111, 102, 58, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 52, 0, 0])]
2015-12-05 06:25:48,593 | ERROR | ool-13-thread-27 | CompositionManager               | 160 - core.caos - 1.1.0.SNAPSHOT | Could not handle incoming message due to configuration error Waiting for modules to connect. MissingModule [Name=SimpleSwitch,loaderIdentification=simple_switch.py,CallCondition=null]: OpenFlowMessage [Header=MessageHeader [Version=VERSION_1_4,Type=OPENFLOW,Length=99,ModuleId=0,TransactionId=0,DatapathId=3],Type=PACKET_IN,OFMessage=OFPacketInVer10(xid=0, bufferId=4294967295, totalLen=81, inPort=2, reason=ACTION, data=[-1, -1, -1, -1, -1, -1, -34, -83, -66, -17, -70, 17, -119, 66, 2, 7, 4, 0, 0, 0, 0, 0, 4, 4, 5, 2, 0, 0, 0, 1, 6, 2, 0, 120, -2, 18, -92, 35, 5, 1, 79, 78, 79, 83, 32, 68, 105, 115, 99, 111, 118, 101, 114, 121, -2, 23, -92, 35, 5, 2, 111, 102, 58, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 52, 0, 0])]

Thank you!

**Comment by schwabe** [comment]
```
 2015-12-05 06:25:48,593 | ERROR | ool-13-thread-27 | CompositionManager | 160 - core.caos - 1.1.0.SNAPSHOT | Could not handle incoming message due to configuration error Waiting for modules to connect. MissingModule [Name=SimpleSwitch,loaderIdentification=simple_switch.py,CallCondition=null]: OpenFlowMessage [Header=MessageHeader [Version=VERSION_1_4,Type=OPENFLOW,Length=99,ModuleId=0,TransactionId=0,DatapathId=3],Type=PACKET_IN,OFMessage=OFPacketInVer10(xid=0, bufferId=4294967295, totalLen=81, inPort=2, reason=ACTION, data=[-1, -1, -1, -1, -1, -1, -34, -83, -66, -17, -70, 17, -119, 66, 2, 7, 4, 0, 0, 0, 0, 0, 4, 4, 5, 2, 0, 0, 0, 1, 6, 2, 0, 120, -2, 18, -92, 35, 5, 1, 79, 78, 79, 83, 32, 68, 105, 115, 99, 111, 118, 101, 114, 121, -2, 23, -92, 35, 5, 2, 111, 102, 58, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 52, 0, 0])]
```

The core is missing the SimpleSwitch. You can check connected modules with ```netide:listmodules```

**Comment by Maoyuxing** [comment]
Is SimpleSwich a module or something? Where is it? How could I connect it to the core?
Very looking forwarding for your help. Thank you!

**Comment by schwabe** [comment]
Your compositon specification specifies the modules. SimpleSwitch the id of backend that has been specified in your composition. See the netide:showcompsition and netide:loadcomposition commands (and the already mentioned listmodules command)

**Comment by ElisaRojas** [comment]
@Maoyuxing The core needs some information on the backend and applications running on top of it. If you want to run a simple switch app in Floodlight, over the core, over ONOS, then you need to specify it in a composition file to the core with the ```netide:loadcomposition``` command that @schwabe mentioned. After it, and when everything is running, if you run the command ```netide:listmodules``` you should see the Floodlight backend and the app running on it. Right now, if you execute ```netide:listmodules```, what is the output you see? :)

**Comment by Maoyuxing** [comment]
Hi, I added OneSimpleSwitch.xml to the core using netide:loadcomposition command. The bankend is ryu-backend with simpleswitch running on top of it. Everything went well. The PACKET_IN and PACKET_OUT packets are both captured. However, if I want to use floodlight-backend with learningswitch app running on top of it. I found there is no related XML file. Should I write it by myself? I tried to write one as follows:

?xml version="1.0" encoding="utf-8"?
CompositionSpecification  xmlns="http://netide.eu/schemas/compositionspecification/v1"
  Modules
      Module id="learningswitch" loaderIdentification="learningswitch.java"/
  /Modules
  Composition
   ModuleCall module="learninglwitch"/
  /Composition
/CompositionSpecification

I omit the '<' and '>' here. It did not work. What should I do? Thank you!
